### PR TITLE
AMQPS support

### PR DESCRIPTION
**Pull request by justinhillsjohnson**

### CODE REVIEW

**Comment by justinhillsjohnson** [comment]
@RichardKnop 

Please take a look at this PR as it provides support for AMQPS. I ran into issues moving our Golang app into production on Aptible and this monkey patch fixes the problem.

On the client side, I implemented TLS using `Config` as such:

```
	conf := &tls.Config{
		InsecureSkipVerify: true,
	}
	return &config.Config{
		Broker:             env.MqBroker,
		ResultBackend:      env.MqResultBackend,
		Exchange:           env.MqExchange,
		ExchangeType:       env.MqExchangeType,
		DefaultQueue:       env.MqDefaultQueue,
		BindingKey:         env.MqBindingKey,
		MaxWorkerInstances: env.MqMaxWorkerInstances,
		TLSConfig:          conf,
	}
```

However, please note that tests are failing in your `master` branch due to unrelated changes in this PR. I can rebase against master once those issues are resolved.

File: v1/factories.go [hunk 1]
```diff
@@ -12,12 +12,16 @@ import (
...
 // BrokerFactory creates a new object of brokers.Interface
-// Currently only AMQP broker is supported
+// Currently only AMQP/S broker is supported
 func BrokerFactory(cnf *config.Config) (brokers.Interface, error) {
 	if strings.HasPrefix(cnf.Broker, "amqp://") {
 		return brokers.NewAMQPBroker(cnf), nil
 	}
 
+	if strings.HasPrefix(cnf.ResultBackend, "amqps://") {
+		return backends.NewAMQPBackend(cnf), nil
```
Comment (RichardKnop) [inline v1/factories.go]: This line should be:

```go
return brokers.NewAMQPBroker(cnf), nil
```

That's why CI was failing. I have merged the PR, fixed this and pushed to master.

Comment (justinhillsjohnson) [inline v1/factories.go]: @RichardKnop Thanks!

Comment (RichardKnop) [inline v1/factories.go]: Also rebase against master as there have been quite a lot of changes in last couple of weeks. After months of low activity I finally managed to get some free time last two weeks. So lots of refactoring and changes in the code. Plus new features.

**Pull request merged**